src/deep_links/mod.rs

```diff
@@ -41,6 +41,86 @@ pub struct ExternalPlayerLink {
+                Some(player_type) => match player_type.as_str() {
+                    "choose" => Some(OpenPlayerLink {
+                        android: Some(format!(
+                            "{}#Intent;type=video/any;scheme=https;end",
+                            http_regex.replace(url, "intent://"),
+                        )),
+                        ..Default::default()
+                    }),
+                    "vlc" => Some(OpenPlayerLink {
+                        ios: Some(format!("vlc-x-callback://x-callback-url/stream?url={url}")),
+                        android: Some(format!(
+                            "{}#Intent;package=org.videolan.vlc;type=video;scheme=https;end",
+                            http_regex.replace(url, "intent://"),
+                        )),
+                        ..Default::default()
+                    }),
+                    "mxplayer" => Some(OpenPlayerLink {
+                        android: Some(format!(
+                            "{}#Intent;package=com.mxtech.videoplayer.ad;type=video;scheme=https;end",
+                            http_regex.replace(url, "intent://"),
+                        )),
+                        ..Default::default()
+                    }),
+                    "justplayer" => Some(OpenPlayerLink {
+                        android: Some(format!(
+                            "{}#Intent;package=com.brouken.player;type=video;scheme=https;end",
+                            http_regex.replace(url, "intent://"),
+                        )),
+                        ..Default::default()
+                    }),
+                    "outplayer" => Some(OpenPlayerLink {
+                        ios: Some(format!("{}", http_regex.replace(url, "outplayer://"))),
+                        ..Default::default()
+                    }),
+                    "infuse" => Some(OpenPlayerLink {
+                        ios: Some(format!("infuse://x-callback-url/play?url={url}")),
+                       ..Default::default()
+                    }),
+                    _ => None,
+                },
+                None => None,
+            },
+            None => None,
+        };
+        let (android_tv, tizen, webos) = match &stream.source {
+            StreamSource::External {
+                android_tv_url,
+                tizen_url,
+                webos_url,
+                ..
+            } => (
+                android_tv_url.as_ref().map(|url| url.to_string()),
+                tizen_url.to_owned(),
+                webos_url.to_owned(),
+            ),
+            _ => (None, None, None),
+        };
+        ExternalPlayerLink {
+            href,
+            download,
+            streaming,
+            open_player,
+            android_tv,
+            tizen,
+            webos,
+            file_name,
+        }
+    }
+}
+
 impl From<(&Stream, &Option<Url>, &Settings)> for ExternalPlayerLink {
```

Comment:
- [ ] Should be removed if we don't want to be able to override the `streaming_url`